ipfs-cluster-service/main.go

```diff
@@ -251,17 +252,114 @@ configuration.
+				{
+					Name:  "export",
+					Usage: "save the IPFS Cluster state to a json file",
+					Description: `
+This command reads the current cluster state and saves it as json for 
+human readability and editing.  Only state formats compatible with this
+version of ipfs-cluster-service can be exported.  By default this command
+prints the state to stdout.
+`,
+					Flags: []cli.Flag{
+						cli.StringFlag{
+							Name:  "file, f",
+							Value: "",
+							Usage: "sets an output file for exported state",
+						},
+					},
+					Action: func(c *cli.Context) error {
+						err := locker.lock()
+						checkErr("acquiring execution lock", err)
+						defer locker.tryUnlock()
+
+						err = export(c.String("file"))
+						checkErr("exporting state", err)
+						return nil
+					},
+				},
+				{
+					Name:  "import",
+					Usage: "load an IPFS Cluster state from an exported state file",
+					Description: `
+This command reads in an exported state file storing the state as a persistent
+snapshot to be loaded as the cluster state when the cluster peer is restarted.
+If an argument is provided, cluster will treat it as the path of the file to
+import.  If no argument is provided cluster will read json from stdin
+`,
+					Action: func(c *cli.Context) error {
+						err := locker.lock()
+						checkErr("acquiring execution lock", err)
+						defer locker.tryUnlock()
+
+						if !c.GlobalBool("force") {
+							logger.Info("Importing new state. All cluster peers must import the same state to avoid raft errors")
+							if !yesNoPrompt("The peer's state will be replaced.  Existing pins may be lost.  Continue? [y/n]") {
+								return nil
+							}
+						}
+
+						// Get the importing file path
+						importFile := c.Args().First()
+						var r io.ReadCloser
+						if importFile == "" {
+							r = os.Stdin
+						} else {
+							r, err = os.Open(importFile)
+							checkErr("reading import file", err)
+						}
+						defer r.Close()
+						err = stateImport(r)
+						checkErr("importing state", err)
+						return nil
+					},
+				},
+				{
+					Name:  "cleanup",
+					Usage: "cleanup persistent consensus state so cluster can start afresh",
+					Description: `
+This command removes the persistent state that is loaded on startup to determine this peer's view of the
+cluster state.  While it removes the existing state from the load path, one invocation does not permanently remove
+this state from disk.  This command renames cluster's data folder to <data-folder-name>.old.0, and rotates other
+deprecated data folders to <data-folder-name>.old.<n+1>, etc for some rotation factor before permanatly deleting 
+the mth data folder (m currently defaults to 5)
+`,
+					Action: func(c *cli.Context) error {
+						err := locker.lock()
+						checkErr("acquiring execution lock", err)
+						defer locker.tryUnlock()
+
+						if !c.GlobalBool("force") {
+							logger.Info("Cleaning up cluster state.  All other peers must also cleanup to avoid raft errors")
```

Comment:
Warning

Reply:
Actually, should only log this when the operation is going to happen (after this if block) or has happened.

Should not say that other peers must also be cleaned up. It may confuse the user. Maybe a single peer needs to be cleaned up.

Perhaps a `the %datafolder  folder has been rotated. Next start will use an empty state.` suffices at the end.